### PR TITLE
:Plug throws error for invalid option

### PR DESCRIPTION
**Pull request by janlazo**
Related: #930, #936

### CODE REVIEW

File: plug.vim [hunk 1]
```diff
@@ -632,17 +632,25 @@ function! plug#(repo, ...)
     let g:plugs[name] = spec
     let s:loaded[name] = get(s:loaded, name, 0)
   catch
-    return s:err(v:exception)
+    return s:err(repo . ' ' . v:exception)
```
Comment (janlazo) [inline plug.vim:635]: Hard to debug which `:Plug` caused the error without this.

**Comment by janlazo** [comment]
@junegunn Any issues with the implementation?

**Comment by junegunn** [comment]
Looks good to me.

1. We can also verify `do` option value.
2. Does this affect the startup time? Would it be negligible given that we have ten to twenty plugins with dictionary options?

**Comment by janlazo** [comment]
1. `Verifying 'do' at `Plug` breaks existing tests. Expected error message is different from the error format I am using for the new tests I added.

2. Is 2ms negligible? I tested with my config (about 50 plugins, 9 plugins passing a dictionary) and I don't notice a slowdown.

File: test/workflow.vader [hunk 2]
```diff
@@ -1055,15 +1059,13 @@ Execute (Post-update hook output; success and failure):
...
-Execute (Post-update hook output; invalid type or funcref):
+Execute (Post-update hook output; invalid funcref):
   call plug#begin()
-  Plug 'junegunn/vim-easy-align', { 'do': 1 }
   Plug 'junegunn/vim-pseudocl',   { 'do': function('call') }
   call plug#end()
 
   silent PlugInstall! 1
   AssertEqual 'x Post-update hook for vim-pseudocl ... Vim(call):E119: Not enough arguments for function: call', getline(5)
-  AssertEqual 'x Post-update hook for vim-easy-align ... Invalid hook type', getline(6)
```
Comment (janlazo) [inline test/workflow.vader:1046]: `Invalid hook type` can happen if the user updates `g:plugs` after `plug#end` and before `PlugInstall`.

**Comment by junegunn** [comment]
```sh
#!/bin/bash

niter=100

rm -f plug.log

measure() {
  for _ in $(seq 1 $niter); do
    vim --startuptime _.log +q
    tail -1 _.log >> plug.log
  done
  cat plug.log
  awk '{ sum += $1 } END { printf "mean: %dms\n", sum / NR }' < plug.log
}

measure
```

Thanks, I also see no noticeable difference in startup time before and after the change.

**Pull request merged**